### PR TITLE
build: create Bazel setup for aio example cli apps

### PR DESCRIPTION
**Pull request by josephperrott**
Written based on work  by @devversion

This PR creates a couple Bazel rules for AIO example apps to use, one defining how to spin up the devserver and another to run the apps e2e tests.

These e2e tests will automatically be included in our CI runs in our `test_ivy_aot` and `test` jobs.

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
From a code point of view (not being an expert in Bazel) this looks good.
What I think is lacking is a lot more documentation. We need to make it very easy for developers of examples to be able to write these `BAZEL.build` files and understand how all the pieces fit together.

**Review by devversion — APPROVED** [review]
LGTM for the bazel stuff :smile: I agree that we should do a bit more documentation. At least a comment for the `aio_example` macro would be good I think.

File: packages/private/testing/BUILD.bazel [hunk 1]
```diff
@@ -1,4 +1,5 @@
 package(default_visibility = [
+    "//aio:__subpackages__",
```
Comment (gkalpak) [inline packages/private/testing/BUILD.bazel]: Why is this necessary?

File: aio/content/examples/aio_example.bzl [hunk 2]
```diff
@@ -0,0 +1,76 @@
+load("//packages/bazel:index.bzl", "protractor_web_test_suite")
```
Comment (gkalpak) [inline aio/content/examples/aio_example.bzl]: I think [aio/tools/examples/][1] (or a subdirectory there) would be a more appropriate place for this file. Would that be a problem (i.e. could files in `aio/content/` be able to "import" it without too much extra complexity)?

[1]: https://github.com/angular/angular/blob/961d663fbef4d6ea2ebb5a60f38799a273fa71ed/aio/tools/examples/README.md

File: aio/content/examples/aio_example.bzl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+# Macro rule for creating the e2e testing target for an aio example.
+def aio_example_e2e(name, srcs, server, deps = []):
+    ts_library(
+        name = "%s_lib" % name,
+        testonly = True,
+        srcs = srcs,
+        tsconfig = "//aio/content/examples:tsconfig-e2e.json",
+        deps = [
+            "//packages/private/testing",
```
Comment (gkalpak) [inline aio/content/examples/aio_example.bzl]: Why is this needed?

File: .bazelignore [hunk 4]
```diff
@@ -1,7 +1,6 @@
 .git
 node_modules
 dist
-aio/content
```
Comment (gkalpak) [inline .bazelignore]: OOC, what consequences does this have?

**Review by gkalpak — COMMENTED** [review]
I don't understand everything (bazel who?), but I don't see anything suspicious 👍 

It will be a little tedious to have a `BAZEL.build` file for each example (not all examples are the same - e.g. some examples might need to exclude specific TS files, etc.) and a little more complicated for contributors (as if it wasn't complicated already 😛), but I guess there is not much we can do about it 🤔

File: aio/content/examples/aio_example.bzl (hunk 3 above)
Comment (josephperrott) [reply aio/content/examples/aio_example.bzl]: This package includes the logic that allows us to flip between Ivy and VE using the `--config` flags in Bazel as described [here](https://github.com/angular/angular/blob/master/docs/BAZEL.md#various-flags-used-for-tests)

File: .bazelignore (hunk 4 above)
Comment (josephperrott) [reply .bazelignore]: In the past we ignore `aio/content` from the perspective of bazel.  Previously if you did anything that attempted to include files from aio Bazel would ignore items under the directory structures.  Now Bazel is given visibility to them.

File: aio/content/examples/aio_example.bzl (hunk 2 above)
Comment (josephperrott) [reply aio/content/examples/aio_example.bzl]: Agreed, done.

File: packages/private/testing/BUILD.bazel (hunk 1 above)
Comment (josephperrott) [reply packages/private/testing/BUILD.bazel]: This line allows for the BUILD rules under `aio/` to depend on `//packages/private/testing`
